### ISSUE TITLE
Mass cannot compute avg_pulses_auto_masks

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#27]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/27) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

I don't know what's going on here, but it has something to do with the difference between the number of channels, and the number of *good* channels.  It also has something to do with the stupid way that we compute multiple "average pulses" per channel (historically, this is rooted in our CDM-compatible origins of Mass).

The best, cleanest way to fix this bug is probably to simplify so that each channel has *one average pulse*, which is what we always assume downstream anyway.

```
#!python

We can skip segment   27
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-14-0119031d9a25> in <module>()
----> 1 data.avg_pulses_auto_masks()

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mass/core/channel_group.pyc in avg_pulses_auto_masks(self, max_pulses_to_use, forceNew)
   1423             if len(m) > max_pulses_to_use:
   1424                 m[max_pulses_to_use:] = False
-> 1425         self.compute_average_pulse(masks, forceNew=forceNew)
   1426 
   1427     def drift_correct(self, forceNew=False, category=None):

/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mass/core/channel_group.pyc in compute_average_pulse(self, masks, subtract_mean, forceNew)
   1026                     average_pulses[imask, :] -= np.mean(average_pulses[imask,
   1027                                                                        :self.nPresamples - ds.pretrigger_ignore_samples])
-> 1028             ds.average_pulse[:] = average_pulses[ichan, :]
   1029 
   1030     def plot_average_pulses(self, channum=None, axis=None, use_legend=True):

IndexError: index 103 is out of bounds for axis 0 with size 103
> /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/mass/core/channel_group.py(1028)compute_average_pulse()
   1027                                                                        :self.nPresamples - ds.pretrigger_ignore_samples])
-> 1028             ds.average_pulse[:] = average_pulses[ichan, :]
   1029 

```

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

If someone other than me wishes to fix this, I suggest you pull from and push to branch:develop.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

TESGroup.avg_pulses_auto_masks now tries to average only on good channels.

The loop used to try all channels, but the mask set was only the size of the
good channels, so it was an error. Fixes #27.
